java/server/src/org/openqa/selenium/netty/server/Http2RequestConverter.java

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.netty.server;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.openqa.selenium.remote.http.HttpMethod;
+import org.openqa.selenium.remote.http.HttpRequest;
+
+public class Http2RequestConverter extends SimpleChannelInboundHandler<HttpObject> {
```

Comment:
This is mirroring the functionality of the `RequestConverter`, but requires http object aggregation in order to work, and so is likely to lead to problems when there's very little memory. In addition, it duplicates logic from that class, making fixing discovered issues more work and opening us up to forgetting to change both places.

I'd suggest deleting this and using `RequestConverter`

Reply:
Thank you for pointing it out. Since Http/2 uses frames with stream ids, I am leveraging to Http2StreamFrameToHttpObjectCodec get the HttpObject to pass it to Selenium handlers. I have moved the Http/2 request handling to `RequestConverter`.